isotovideo

```diff
@@ -129,160 +114,41 @@ my $return_code = 1;
-checkout_git_repo_and_branch('NEEDLES_DIR');
-
-bmwqemu::ensure_valid_vars();
-
-# as we are about to load the test modules checkout the specified git refspec,
-# if specified, or simply store the git hash that has been used. If it is not a
-# git repo fail silently, i.e. store an empty variable
-
-$bmwqemu::vars{TEST_GIT_HASH} = checkout_git_refspec($bmwqemu::vars{CASEDIR} => 'TEST_GIT_REFSPEC');
-
-# set a default distribution if the tests don't have one
-$testapi::distri = distribution->new;
-
-$bmwqemu::vars{WHEELS_DIR} ||= $bmwqemu::vars{CASEDIR};
-checkout_wheels($bmwqemu::vars{WHEELS_DIR});
-load_test_schedule;
-
-# start the command fork before we get into the backend, the command child
-# is not supposed to talk to the backend directly
-($cmd_srv_process, $cmd_srv_fd) = commands::start_server($cmd_srv_port = $bmwqemu::vars{QEMUPORT} + 1);
+diag(_get_version_string());
 
-testapi::init();
-needle::init();
-bmwqemu::save_vars();
+# enable debug default when started from a tty
+$log::direct_output = $options{debug};
 
-my $testfd;
-($testprocess, $testfd) = autotest::start_process();
+$bmwqemu::scriptdir = $installprefix;
 
-$backend = OpenQA::Isotovideo::Backend->new;
+my $runner = OpenQA::Isotovideo::Runner->new;
+$runner->_init_bmwqemu(@ARGV);
+$runner->prepare;
+$runner->start_autotest;
+$runner->create_backend;
 
 spawn_debuggers;
 
-# stop main loop as soon as one of the child processes terminates
-my $stop_loop = sub (@) { $command_handler->loop(0) if $command_handler->loop; };
-$testprocess->once(collected => $stop_loop);
-$backend->process->once(collected => $stop_loop);
-$cmd_srv_process->once(collected => $stop_loop);
-
-$command_handler = OpenQA::Isotovideo::CommandHandler->new(
-    cmd_srv_fd => $cmd_srv_fd,
-    test_fd => $testfd,
-    backend_fd => $backend->process->channel_in,
-    backend_out_fd => $backend->process->channel_out,
-);
-$command_handler->on(tests_done => sub (@) {
-        CORE::close($testfd);
-        $testfd = undef;
-        stop_autotest;
-});
-$command_handler->on(signal => sub ($event, $sig) {
-        $backend->stop if defined $backend;    # uncoverable statement
-        stop_commands("received signal $sig");    # uncoverable statement
-        stop_autotest;    # uncoverable statement
-        _exit(1);    # uncoverable statement
-});
-$command_handler->setup_signal_handler;
+$runner->handle_commands;
 
 $return_code = 0;
 
 # enter the main loop: process messages from autotest, command server and backend
-$command_handler->run;
+$runner->run;
 
 # terminate/kill the command server and let it inform its websocket clients before
-stop_commands('test execution ended');
+$runner->stop_commands('test execution ended');
 
-if ($testfd) {
+if ($runner->testfd) {
     $return_code = 1;    # unusual shutdown
-    CORE::close $testfd;
-    stop_autotest;
+    CORE::close $runner->testfd;
+    $runner->stop_autotest();
```

Comment:
I guess these two lines should also have `# uncoverable statement` markers

Reply:
done